### PR TITLE
[JSX] Don't add extra newline at top or bottom (fixes #619)

### LINKED ISSUE #619: [JSX] - should prettier keep blank lines after opening and before closing tags?
Prettier removes blank lines at the start and at the end of the blocks:

In
```javascript
const func = () => {
  
  
  const foo = 1
  
  
  const bar = 2
  
  
  const baz = 3
  
  
}

if (true) {
    
  
  const foo = 1
  
  
  const bar = 2
  
  
  const baz = 3
  
  
}
```
Out
```javascript
const func = () => {
  const foo = 1;

  const bar = 2;

  const baz = 3;
};

if (true) {
  const foo = 1;

  const bar = 2;

  const baz = 3;
}
```

But keeps extra one line in case of JSX tags:

In
```javascript
const comp = (

  
<div>

  	<div>
  foo
  </div>
  
  
  
  	<div>
  bar
  </div>
  
  	<div>
  baz
  </div>
  
  </div>

)
```

Out
```javascript
const comp = (
  <div>

    <div>
      foo
    </div>

    <div>
      bar
    </div>

    <div>
      baz
    </div>

  </div>
);


```

[Example.](https://jlongster.github.io/prettier/#%7B%22content%22%3A%22const%20comp%20%3D%20(%5Cn%5Cn%20%20%5Cn%3Cdiv%3E%5Cn%5Cn%20%20%5Ct%3Cdiv%3E%5Cn%20%20foo%5Cn%20%20%3C%2Fdiv%3E%5Cn%20%20%5Cn%20%20%5Cn%20%20%5Cn%20%20%5Ct%3Cdiv%3E%5Cn%20%20bar%5Cn%20%20%3C%2Fdiv%3E%5Cn%20%20%5Cn%20%20%5Ct%3Cdiv%3E%5Cn%20%20baz%5Cn%20%20%3C%2Fdiv%3E%5Cn%20%20%5Cn%20%20%3C%2Fdiv%3E%5Cn%5Cn)%5Cn%5Cn%5Cnconst%20func%20%3D%20()%20%3D%3E%20%7B%5Cn%20%20%5Cn%20%20const%20foo%20%3D%201%5Cn%20%20%5Cn%20%20%5Cn%20%20const%20bar%20%3D%202%5Cn%20%20%5Cn%20%20%5Cn%20%20const%20baz%20%3D%203%5Cn%20%20%5Cn%20%20%5Cn%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3Afalse%2C%22bracketSpacing%22%3Atrue%2C%22doc%22%3Afalse%7D%7D)

### PR DESCRIPTION
**Pull request by rattrayalex**
fixes #619

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -2385,32 +2385,20 @@ function printJSXElement(path, options, print) {
   let forcedBreak = false;
 
   // Trim trailing lines, recording if there was a hardline
-  let numTrailingHard = 0;
   while (children.length && isLineNext(util.getLast(children))) {
-    if (willBreak(util.getLast(children))) {
-      ++numTrailingHard;
+    if (!forcedBreak && willBreak(util.getLast(children))) {
```
Comment (rattrayalex) [inline src/printer.js]: `!forcedBreak` added as minor perf optimization

**Comment by rattrayalex** [comment]
ready for review @jlongster 
(do you wish to be notified in this way?)

File: tests/jsx-newlines/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -95,21 +95,17 @@ newlines_text = (
 
 newlines_text_spaced = (
   <div>
-
     space above
 
     space below
-
```
Comment (lydell) [inline tests/jsx-newlines/__snapshots__/jsfmt.spec.js.snap]: Isn't saying "space above" and "space below" a bit misleading now? ;)

Comment (rattrayalex) [inline tests/jsx-newlines/__snapshots__/jsfmt.spec.js.snap]: 😉 

I thought about changing that, actually, but figured it wouldn't be worth the diff obfuscation of "this change removed the spaces above and below"

**Comment by jlongster** [comment]
Let's keep the spaces for now, as discussed in https://github.com/jlongster/prettier/issues/619. Thanks for implementing this anyway!

**Pull request closed without merging**